src/main/java/com/commercetools/sync/producttypes/utils/ProductTypeUpdateAttributeDefinitionActionUtils.java

```diff
@@ -158,15 +159,44 @@
                 final AttributeDefinitionDraft matchingNewAttributeDefinitionDraft =
                     newAttributeDefinitionDraftsNameMap.get(oldAttributeDefinitionName);
                 return ofNullable(matchingNewAttributeDefinitionDraft)
-                    .map(attributeDefinitionDraft ->
-                        buildActions(oldAttributeDefinition, attributeDefinitionDraft)
-                    )
+                    .map(attributeDefinitionDraft -> {
+                        // attribute type is required so if null we let commercetools to throw exception
+                        if (attributeDefinitionDraft.getAttributeType() != null) {
+                            if (haveSameAttributeType(oldAttributeDefinition, attributeDefinitionDraft)) {
+                                return buildActions(oldAttributeDefinition, attributeDefinitionDraft);
+                            } else {
+                                return Arrays.asList(
+                                    RemoveAttributeDefinition.of(oldAttributeDefinitionName),
+                                    AddAttributeDefinition
```

Comment:
ah ok, I see you doing it here :)

Reply:
I think it would be important to put in the Javadoc of the method and/or the documentation of the productType sync.

Reply:
7a196a6c573acbde9603e1046cb35d06b450626c